File: Hash Table/202. Happy Number/happy.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()

        while n != 1:
            if n in seen:
                break
            sum = 0
            seen.add(n)

            while n:
                rem = n % 10
                sum += rem * rem
                n //= 10
            n = sum

        return n == 1

File: Hash Table/202. Happy Number/happy.cpp
class Solution
{
public:
    bool isHappy(int n)
    {
        unordered_set<int> seen;

        while (n != 1 && seen.find(n) == seen.end())
        {
            int sum = 0;
            seen.insert(n);
            while (n)
            {
                int rem = n % 10;
                sum += rem * rem;
                n /= 10;
            }

            n = sum;
        }
        return n == 1;
    }
};